packages/ni.measurementlink.sessionmanagement.v1.client/src/ni/measurementlink/sessionmanagement/v1/client/_helpers.py

```diff
@@ -0,0 +1,64 @@
+import socket
+
+import win32api
+
+from ni.measurementlink.sessionmanagement.v1.client._constants import (
+    REGISTERED_HOSTNAME,
+    REGISTERED_IPADDRESS,
+    REGISTERED_USERNAME,
+    RESERVED_HOSTNAME,
+    RESERVED_IPADDRESS,
+    RESERVED_USERNAME,
+)
+
+
+def get_machine_details() -> tuple[dict[str, str], dict[str, str]]:
+    """Get the machine details for reserved and registered annotations."""
+    hostname = _get_hostname()
+    username = _get_username()
+    ip_address = _get_ip_address(hostname)
+
+    reserved = {
+        RESERVED_HOSTNAME: hostname,
+        RESERVED_USERNAME: username,
+        RESERVED_IPADDRESS: ip_address,
+    }
+
+    registered = {
+        REGISTERED_HOSTNAME: hostname,
+        REGISTERED_USERNAME: username,
+        REGISTERED_IPADDRESS: ip_address,
+    }
+
+    return reserved, registered
+
+
+def remove_reservation_annotations(annotations: dict[str, str]) -> dict[str, str]:
+    """Remove reserved annotations from the provided annotations."""
+    reservation_keys = {
+        RESERVED_HOSTNAME,
+        RESERVED_USERNAME,
+        RESERVED_IPADDRESS,
+    }
+    return {k: v for k, v in annotations.items() if k not in reservation_keys}
+
+
+def _get_hostname() -> str:
+    try:
+        return win32api.GetComputerName()
+    except Exception:
+        return ""
+
+
+def _get_username() -> str:
+    try:
+        return win32api.GetUserName()
+    except Exception:
+        return ""
```

Comment:
Don't sweep all errors under the rug.

Why would these fail?
- Something is really wrong -> we should report that to the user, not hide the error
- You're running this code on Linux or Mac -> add `if sys.platform == "win32"`. The else case should raise NotImplementedError, like in https://github.com/ni/ni-apis-python/blob/main/packages/ni.measurementlink.discovery.v1.client/src/ni/measurementlink/discovery/v1/client/_support.py

Reply:
Ok. I assume that logic is in the UI.